src/plugins/keras_support/elu_operation.cpp

```diff
@@ -0,0 +1,207 @@
+#include <blaze_tensor/Math.h>
+#endif
+
+///////////////////////////////////////////////////////////////////////////////
+namespace phylanx { namespace execution_tree { namespace primitives
+{
+    match_pattern_type const elu_operation::match_data =
+    {
+         match_pattern_type{
+            "elu",
+            std::vector<std::string>{"elu(_1, _2)"},
+            &create_elu_operation,
+            &create_primitive<elu_operation>, R"(
+            a
+            Args:
+
+                a (array_like) : input array
+                alpha (optional, number) : scale for the negative factor
+
+            Returns:
+
+            Returns an array of the same shape that is the exponential linear
+            unit of the input and is defined as:
+            * f(x) = alpha * (exp(x) - 1.) for x < 0,
+            * f(x) = x for x >= 0.)"
+        }
+    };
+
+    elu_operation::elu_operation(primitive_arguments_type&& operands,
+        std::string const& name, std::string const& codename)
+        : primitive_component_base{ std::move(operands), name, codename }
+    {}
+
+    primitive_argument_type elu_operation::elu0d(mat_type&& arg,
+        double alpha) const
+    {
+        //  ELU activation function
+        auto elu_ = [alpha](auto&& x)
+        {
+            //  NB : Might be unusable with the upcoming CUDA version of blaze
+            //  because lambda kernels are wrapped inside std::function objects.
+            //  See: https://devblogs.nvidia.com/new-compiler-features-cuda-8/,
+            //  then find "There's one caveat:"
+
+            //  NB2: Keeping all that code as a single expression may suggest
+            //  the compiler to use only float operations instead of branching,
+            //  also it allows optimizations with libraries like boost_simd that
+            //  provide comparison operators that directly translate into SIMD.
+            //  Using an if statement instead makes vectorization harder if not
+            //  impossible.
+
+            return (x >= 0.) * ( x )
+                 + (x <  0.) * ( alpha * (std::exp(x) - 1.) );
+        };
+
+        //  NB : Enforcing storageXd_type constructors here otherwise the
+        //  call for primitive_argument_type's constructor would be ambiguous.
+        //  These correspond to 0D/1D/2D/3D types in Blaze and thus should have
+        //  costless move constructors.
+
+        return primitive_argument_type{mat_type::storage0d_type{
+            elu_(arg.scalar()) }};
+    }
+
+    primitive_argument_type elu_operation::elu1d(mat_type&& arg,
+        double alpha) const
+    {
+        auto elu_ = [alpha](auto&& x)
+        {
+            return (x >= 0.) * ( x )
+                 + (x <  0.) * ( alpha * (std::exp(x) - 1.) );
+        };
+
+        return primitive_argument_type{mat_type::storage1d_type{
+            arg.is_ref() ? blaze::map(arg.vector(), elu_)
+                : blaze::map(std::move(arg.vector()), elu_)
+        }};
```

Comment:
Are you sure that `blaze::map` essentially turns into an in-place operation when used on rvalues? … interesting, I was not aware of this.

Reply:
I looked it up, it only accepts `const&` parameters. Do you want me to remove that ?

Reply:
We should do different things as if `arg.is_ref()` we should allocate new memory, otherwise the operation should be performed in-place. `blaze::map` always allocates new memory, IIRC.